### ISSUE TITLE
SSL object weirdness when not inside of cqueues

### ISSUE THREAD

**Issue by daurnimator** [wahern/cqueues#30]
The following example only works if you uncomment the lines to place the operations in a cqueues coroutine.
```lua
local cqs = require "cqueues"
local cs = require "cqueues.socket"
-- assert(cqs.new():wrap(function()
	local s = cs.connect("google.com", 443)
	s:starttls()
	local ssl = s:checktls()
	print(ssl:getPeerCertificate())
-- end):loop())
```

**Comment by wahern** [comment]
The issue here is that :starttls never originally blocked/polled. It simply put the socket into the TLS handshake state and returned. Later :starttls was fixed to wait until the handshake completed before returning, as part of a refactor to making timeouts more consistent across the API. However, the original example code called :starttls outside an event loop, so I added a check to preserve the original behavior. So in your code :starttls is returning immediately, and :getPeerCertificate returns nothing because the handshake hasn't completed.

I suppose now that cqueues.poll has been hacked to support polling outside of an event loop I could simply remove that code. I'm just worried it might break some code out in the wild by blocking the application unexpectedly. Alternatively, :checktls could be changed to poll on handshake completion before returning.

I suppose the former option improves consistency, and I'll just have to remember to highlight the change.
